src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/oracle/PMTransactionDataProvider.kt

```diff
@@ -0,0 +1,112 @@
+            is SearchTransactionRequestFiscalCodeDto ->
+                Mono.error(RuntimeException("Not implemented yet"))
+            else ->
+                Mono.error(
+                    InvalidSearchCriteriaException(
+                        TransactionDataProvider.SearchTypeMapping.getSearchType(
+                            searchCriteria.javaClass
+                        ),
+                        ProductDto.PM
+                    )
+                )
+        }
+
+    override fun findResult(
+        searchCriteria: HelpDeskSearchTransactionRequestDto,
+        pageSize: Int,
+        pageNumber: Int
+    ): Mono<List<TransactionResultDto>> {
+        val searchCriteriaType =
+            TransactionDataProvider.SearchTypeMapping.getSearchType(searchCriteria.javaClass)
+        val invalidSearchCriteriaError =
+            Mono.error<List<TransactionResultDto>>(
+                InvalidSearchCriteriaException(searchCriteriaType, ProductDto.PM)
+            )
+        return when (searchCriteria) {
+            is SearchTransactionRequestPaymentTokenDto -> invalidSearchCriteriaError
+            is SearchTransactionRequestRptIdDto -> invalidSearchCriteriaError
+            is SearchTransactionRequestTransactionIdDto -> invalidSearchCriteriaError
+            is SearchTransactionRequestEmailDto ->
+                getResultSetFromPaginatedQuery(
+                    resultQuery =
+                        buildTransactionByUserEmailPaginatedQuery(searchCriteria.userEmail),
+                    pageNumber = pageNumber,
+                    pageSize = pageSize,
+                    searchType = searchCriteriaType
+                )
+            is SearchTransactionRequestFiscalCodeDto ->
+                Mono.error(RuntimeException("Not implemented yet"))
+            else -> invalidSearchCriteriaError
+        }
+    }
+
+    private fun getTotalResultCount(totalRecordCountQuery: String): Mono<Int> =
+        Flux.usingWhen(
+                connectionFactory.create(),
+                { connection ->
+                    Flux.from(connection.createStatement(totalRecordCountQuery).execute())
+                        .flatMap { result ->
+                            result.map { row -> row[0, java.lang.Long::class.java]!!.toInt() }
+                        }
+                        .doOnNext { logger.info("Total transaction found: $it") }
+                },
+                { it.close() }
+            )
+            .toMono()
+
+    private fun getResultSetFromPaginatedQuery(
+        resultQuery: String,
+        pageNumber: Int,
+        pageSize: Int,
+        searchType: String
+    ): Mono<List<TransactionResultDto>> =
+        Flux.usingWhen(
+                connectionFactory.create(),
+                { connection ->
+                    val offset = pageNumber * pageSize
+                    val query = resultQuery.format(offset, pageSize)
```

Comment:
Can we use a query builder?

Reply:
What do you suggest? 
Here the aim is the designed query unaffected and just add pagination/where parameters

Reply:
Fix with https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/pull/9/commits/26efdda12958812eb1c9f3a0d470804b62082080